rclcpp_components/src/component_container_mt.cpp

```diff
@@ -21,8 +21,27 @@
 int main(int argc, char * argv[])
 {
   /// Component container with a multi-threaded executor.
-  rclcpp::init(argc, argv);
-  auto exec = std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
+  auto vargv = rclcpp::init_and_remove_ros_arguments(argc, argv);
+  size_t number_of_threads{0};
+  rclcpp::Logger logger{rclcpp::get_logger("component_container_mt")};
+
+  if (!vargv.empty()) {
+    for (auto itr = vargv.begin(); itr != vargv.end() - 1; ++itr) {
+      if (*itr == "--thread-num" || *itr == "-t") {
```

Comment:
arguments that are not `--thread-num/-t` are being skipped, we should instead fail when finding them

Reply:
FIxed at https://github.com/ros2/rclcpp/pull/1693/commits/e0110cc9c77618db584a9f62809a7e093084df35